Modules/ArcGIS/ArcGISUtility.psm1

```diff
@@ -125,8 +129,12 @@ function Wait-ForServiceToReachDesiredState
     if ($Attempts++ -gt 0) {  # to skip the message for first attempt
       Write-Verbose "Checking state of Service '$ServiceName'. Attempt # $Attempts"        
     }    
+    if($MachineName -ne $null -or $MachineName -ne ""){
+        $Service = Get-Service -Name $ServiceName -ErrorAction Ignore -ComputerName $MachineName
```

Comment:
Issue - Needs to be validated if this will work when the resource is running as a Local System Account.